integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/ingest/TableRegisterSupport.java

```diff
@@ -214,12 +191,32 @@ protected String createDatabaseDDL(@Nonnull final String source) {
 
     protected String createDDL(String source, String entity, ColumnSpec[] columnSpecs, ColumnSpec[] partitions, String feedFormatOptions, String targetFormatOptions, String targetTableProperties,
                                TableType tableType) {
-        String tableName = tableType.deriveQualifiedName(source, entity);
-        String partitionSQL = tableType.derivePartitionSpecification(partitions);
-        String columnsSQL = tableType.deriveColumnSpecification(columnSpecs, partitions, feedFormatOptions);
-        String locationSQL = tableType.deriveLocationSpecification(config.pathForTableType(tableType), source, entity);
-        String formatOptionsSQL = tableType.deriveFormatSpecification(feedFormatOptions, targetFormatOptions);
-        String tblPropertiesSQL = tableType.deriveTableProperties(targetTableProperties);
+
+        String tableName;
+        String columnsSQL;
+        String formatOptionsSQL;
+        String partitionSQL;
+        String locationSQL;
+        String tblPropertiesSQL;
+
+        switch (tableType) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this "switch" statement by "if" statements to increase readability. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1301)